lib/charms/tls_certificates_interface/v2/tls_certificates.py

```diff
@@ -1171,15 +1207,19 @@ def _on_relation_changed(self, event: RelationChangedEvent) -> None:
             certificate_creation_request["certificate_signing_request"]
             for certificate_creation_request in provider_certificates
         ]
-        requirer_unit_csrs = [
-            certificate_creation_request["certificate_signing_request"]
+        requirer_unit_certificate_requests = [
+            (
+                certificate_creation_request["certificate_signing_request"],
+                certificate_creation_request.get("ca", False),
+            )
```

Comment:
```suggestion
        requirer_unit_certificate_requests = [
            {
                "csr": certificate_creation_request["certificate_signing_request"],
                "is_ca": certificate_creation_request.get("ca", False),
            }
```

I would suggest using a dict instead of a tuple here to help with readability.